easybuild/easyblocks/s/star_ccm.py

```diff
@@ -0,0 +1,114 @@
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import change_dir
+from easybuild.tools.run import run_cmd
+
+
+class EB_STAR_minus_CCM_plus_(EasyBlock):
+    """Support for building/installing STAR-CCM+."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialise STAR-CCM+ easyblock."""
+        super(EB_STAR_minus_CCM_plus_, self).__init__(*args, **kwargs)
+        self.starccm_subdir = None
+        self.starview_subdir = None
+
+    def configure_step(self):
+        """No configuration procedure for STAR-CCM+."""
+        pass
+
+    def build_step(self):
+        """No build procedure for STAR-CCM+."""
+        pass
+
+    def install_step(self):
+        """Custom install procedure for STAR-CCM+."""
+        cands = glob.glob("./STAR-CCM+%s_*.sh" % self.version)
+        if len(cands) == 1:
+            install_script = cands[0]
+
+            env.setvar('CHECK_DISK_SPACE', 'OFF')
+            env.setvar('IATEMPDIR', tempfile.mkdtemp())
+
+            cmd = ' '.join([
+                install_script,
+                "-i silent",
+                "-DINSTALLDIR=%s" % self.installdir,
+                "-DINSTALLFLEX=false",
+                "-DADDSYSTEMPATH=false",
+            ])
+            run_cmd(cmd, log_all=True, simple=True)
+        else:
+            raise EasyBuildError("Found no or multiple possible matches for STAR-CCM+ installation script: %s", cands)
+
+    def find_starccm_subdirs(self):
+        """Determine subdirectory of install directory in which STAR-CCM+ was installed."""
+        cwd = change_dir(self.installdir)
+        cands = glob.glob(os.path.join(self.version + '*', 'STAR-CCM+%s*' % self.version))
+        if len(cands) == 1:
+            self.starccm_subdir = cands[0]
+            self.log.info("Found STAR-CCM+ subdirectory: %s", self.starccm_subdir)
+            self.starview_subdir = os.path.join(os.path.dirname(self.starccm_subdir), 'STAR-View+%s' % self.version)
+        else:
+            raise EasyBuildError("Failed to determine the STAR-CCM+ subdirectory in %s: %s", self.installdir, cands)
+
+        change_dir(cwd)
+
+    def sanity_check_step(self):
+        """Custom sanity check for STAR-CCM+."""
+        if self.starccm_subdir is None or self.starview_subdir is None:
+            self.find_starccm_subdirs()
+
+        custom_paths = {
+            'files': [os.path.join(self.installdir, self.starccm_subdir, 'star', 'bin', 'starccm+'),
+                      os.path.join(self.installdir, self.starview_subdir, 'bin', 'starview+')],
+            'dirs': [],
+        }
+        super(EB_STAR_minus_CCM_plus_, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_extra(self):
+        """Extra statements specific to STAR-CCM+ to include in generated module file."""
+        if self.starccm_subdir is None or self.starview_subdir is None:
+            self.find_starccm_subdirs()
```

Comment:
is this still needed? Sanity check already does it?

Reply:
It's needed when doing `--module-only --force`, because then the sanity check step is never run.